app/Casts/MoneyCast.php

```
@@ -0,0 +1,108 @@
+    }
+
+    public function get(Model $model, string $key, mixed $value, array $attributes): ?Money
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        return Money::of($value, $this->resolveCurrency($model));
+    }
+
+    public function set(Model $model, string $key, mixed $value, array $attributes): array
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$model'. (undefined)

(UnusedFormalParameter)